### PR TITLE
Python: Only pass metadata that we need

### PR DESCRIPTION
**Pull request by hoodmane**
Rather than serializing the entire workerd config into a JSON string and then deserializing it at runtime this makes another readonly file system so we can copy only the data that Python wants to look at. This will copy all Data, Text, and Python files once at startup into the MetadataReader since I am not sure about the lifespan of the capnp config and it seems easier to understand if we make the module own its data.

### CODE REVIEW

File: src/workerd/api/pyodide/pyodide.h [hunk 1]
```diff
@@ -1,45 +1,185 @@
...
+    JSG_METHOD(getNames);
+    JSG_METHOD(getSizes);
+    JSG_METHOD(read);
+  }
+};
+
+using Worker = server::config::Worker;
+
+inline jsg::Ref<PyodideMetadataReader> makePyodideMetadataReader(Worker::Reader conf) {
+  auto modules = conf.getModules();
```
Comment (jasnell) [inline src/workerd/api/pyodide/pyodide.h:118]: This method is rather too large to be inlined. Would much rather this be moved out into a `pyodide.c++`

Comment (hoodmane) [reply src/workerd/api/pyodide/pyodide.h]: Okay I did this.

File: src/pyodide/python-entrypoint-helper.js [hunk 2]
```diff
@@ -93,89 +94,19 @@ const EMBEDDED_PYTHON_PACKAGES = [
...
 
   if (pythonRequirements.length > 0) {
-    await pyodide.loadPackage(pythonRequirements);
-  }
-
-  if (micropipRequirements.length > 0) {
-    // Micropip and ssl packages are pre-loaded via the packages tarball. This means
-    // we should be able to load micropip directly now.
-
     const micropip = pyodide.pyimport("micropip");
```
Comment (dom96) [inline src/pyodide/python-entrypoint-helper.js:108]:
```suggestion
    // Our embedded packages tarball always contains at least `micropip` so we can load it here safely.
    const micropip = pyodide.pyimport("micropip");
```

File: src/workerd/api/BUILD.bazel [hunk 3]
```diff
@@ -92,16 +96,22 @@ wd_cc_library(
 
 wd_cc_library(
     name = "pyodide",
-    srcs = ["pyodide.c++"],
+    srcs = [
+        "pyodide.c++",
+        "pyodide/pyodide.c++",
```
Comment (dom96) [inline src/workerd/api/BUILD.bazel:101]: These should be merged together, or the second one given a more descriptive name.

**Review by dom96 — APPROVED** [review]

File: src/workerd/api/BUILD.bazel (hunk 3 above)
Comment (hoodmane) [reply src/workerd/api/BUILD.bazel:101]: I'll leave that to a followup. But it would also be good to merge `pyodide.h` and `pyodide/pyodide.h`

**Pull request merged**